packages/tools/src/tools/ScaleOverlayTool.ts

```diff
@@ -0,0 +1,666 @@
+
+    const viewportCanvasCornersInWorld =
+      csUtils.getViewportImageCornersInWorld(viewport);
+
+    let annotation = this.editData.annotation;
+
+    if (!annotation) {
+      const newAnnotation: ScaleOverlayAnnotation = {
+        metadata: {
+          toolName: this.getToolName(),
+          viewPlaneNormal: <Types.Point3>[...viewPlaneNormal],
+          viewUp: <Types.Point3>[...viewUp],
+          FrameOfReferenceUID: viewport.getFrameOfReferenceUID(),
+          referencedImageId: null,
+        },
+        data: {
+          handles: {
+            points: viewportCanvasCornersInWorld,
+          },
+        },
+      };
+
+      addAnnotation(element, newAnnotation);
+      annotation = newAnnotation;
+    } else {
+      this.editData.annotation.data.handles.points =
+        viewportCanvasCornersInWorld;
+    }
+
+    this.editData = {
+      viewport,
+      renderingEngine,
+      annotation,
+    };
+  };
+
+  onSetToolEnabled = (): void => {
+    this._init();
+  };
+
+  onCameraModified = (evt: Types.EventTypes.CameraModifiedEvent): void => {
+    // If the camera is modified, we need to update the viewport
+    // that the camera was modified on
+    this.configuration.viewportId = evt.detail.viewportId;
+    this._init();
+  };
+
+  /**
+   * Used to draw the scale annotation in each request animation
+   * frame.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   * @returns
+   */
+
+  renderAnnotation(
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ) {
+    if (!this.editData.viewport) {
+      return;
+    }
+    const location = this.configuration.scaleLocation;
+    const { annotation, viewport } = this.editData;
+    const image = viewport.getImageData();
+    const imageId = viewport.getCurrentImageId();
+    const canvas = enabledElement.viewport.canvas;
+
+    const renderStatus = false;
+
+    if (!viewport) {
+      return renderStatus;
+    }
+
+    const styleSpecifier: StyleSpecifier = {
+      toolGroupId: this.toolGroupId,
+      toolName: this.getToolName(),
+      viewportId: enabledElement.viewport.id,
+    };
+
+    let rowPixelSpacing = image.spacing[0];
+    let colPixelSpacing = image.spacing[1];
+    const imagePlane = metaData.get('imagePlaneModule', imageId);
+
+    // if imagePlane exists, set row and col pixel spacing
+    if (imagePlane) {
+      rowPixelSpacing =
+        imagePlane.rowPixelSpacing || imagePlane.rowImagePixelSpacing;
+      colPixelSpacing =
+        imagePlane.columnPixelSpacing || imagePlane.colImagePixelSpacing;
+    }
+
+    // Check whether pixel spacing is defined
+    if (!rowPixelSpacing || !colPixelSpacing) {
+      console.warn(
+        `Unable to define rowPixelSpacing or colPixelSpacing from data on ScaleOverlayTool's renderAnnotation`
+      );
+      return;
+    }
```

Comment:
imageId is limited to stackViewport, you are getting imagePlaenModule for the imageId and then row/col pixel spacing. What are you using row/col pixel spacing for? I see it is only used for checking they exists. 

So basically you are limitting this tool for stack viewport for no reason?

Reply:
This section has been removed, I was originally using pixel spacing in calculating the scale size but this section is not of use anymore.